sky/skylet/job_lib.py

```diff
@@ -91,12 +91,16 @@ class JobStatus(enum.Enum):
     SUCCEEDED = 'SUCCEEDED'
     # The job fails due to the user code or a system restart.
     FAILED = 'FAILED'
+    # The job setup failed (--detach-setup). It needs to be placed after
+    # the `FAILED` state, so that the status set by our generated ray
+    # program will not be overwritten by ray's job status (FAILED).
```

Comment:
Can we also document why we need this in addition to FAILED?

Reply:
It is only for a better UX, so that the user can find which part fails for a job quickly. Wdyt?

Reply:
I think it's okay to keep. My previous concern was it creates some more inconsistency:  When --detach-setup is not set, a failing setup script will show on the terminal and not show an entry in `sky queue`; with it set, such a failing setup will have an entry with a new state in the queue. But agreed it's better UX and this may be fine.